framework/CodeInterfaces/Prescient/PrescientCodeInterface.py

```diff
@@ -0,0 +1,228 @@
+      @ In, preExec, string, optional, a string the command that needs to be pre-executed before the actual command here defined
+      @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is a list of commands to run the code (string), returnCommand[1] is the name of the output root
+    """
+    #print(inputFiles, exe, clargs, fargs, preExec)
+    #print("generateCommand")
+    runnerInput = []
+    for inp in inputFiles:
+      if inp.getType() == 'PrescientRunnerInput':
+        runnerInput.append(("parallel", "runner.py "+inp.getAbsFile()))
+
+    return (runnerInput, os.path.join(inputFiles[0].getPath(), "output"))
+
+  def createNewInput(self, inputs, oinputs, samplerType, **Kwargs):
+    """
+      This method is used to generate an input based on the information passed in.
+      @ In, inputs, list,  list of current input files (input files from last this method call)
+      @ In, oinputs, list, list of the original input files
+      @ In, samplerType, string, Sampler type (e.g. MonteCarlo, Adaptive, etc. see manual Samplers section)
+      @ In, Kwargs, dictionary, kwarded dictionary of parameters. In this dictionary there is another dictionary called "SampledVars"
+            where RAVEN stores the variables that got sampled (e.g. Kwargs['SampledVars'] => {'var1':10,'var2':40})
+      @ Out, newInputFiles, list, list of newer input files, list of the new input files (modified and not)
+    """
+    #print(inputs, oinputs, samplerType, Kwargs)
+    #print("createNewInput")
+    self._output_directory = None
+    for singleInput in inputs:
+      if singleInput.getType() == 'PrescientRunnerInput':
+        #print("Need to modify", singleInput, "to fix", prescientLocation)
+        newLines = []
+        for line in open(singleInput.getAbsFile(),"r").readlines():
+          if line.lstrip().startswith("--model-directory=") and prescientLocation is not None:
+            items = line.split("=",1)[1].split("|")
+            newPath = prescientLocation
+            started = False
+            for item in items:
+              if item == "prescient":
+                started = True
+              if started:
+                newPath = os.path.join(newPath, item)
+            line = "--model-directory="+newPath
+          elif line.lstrip().startswith("--output-directory="):
+            self._output_directory = line.split("=",1)[1].rstrip()
+          newLines.append(line)
+        newFile = open(singleInput.getAbsFile(),"w")
+        for line in newLines:
+          newFile.write(line)
+        newFile.close()
+      elif singleInput.getType() == 'PrescientInput':
+        #print("SampledVars", Kwargs["SampledVars"])
+        #print("Modifying", singleInput)
+        data = open(singleInput.getAbsFile(),"r").read()
+        for var in Kwargs["SampledVars"]:
+          data = data.replace("$"+var+"$", str(Kwargs["SampledVars"][var]))
```

Comment:
this is similar to the GenericInterface parser... should we just add this option in there and use the GenericParser here? to avoid redundancy?

Reply:
Hm, I'll just remove this way of doing it, since processData can do $(var)$